roman_photoz/roman_catalog_process.py

```diff
@@ -161,17 +152,26 @@ def format_data(self, cat_data: Table):
         # get information about Roman filters
         bands = self.config["FILTER_LIST"].split(",")
         logger.debug(f"Found {len(bands)} bands in filter list")
+        # get the filter IDs for which we have data
+        observed_filter_ids = sorted(
+            {x.split("_")[1] for x in cat_data.colnames if "flux" in x}
+        )
+        # get a list of all the filter IDs
+        all_filter_ids = sorted([x.split("_")[-1].split(".")[0].lower() for x in bands])
 
+        self.data["label"] = cat_data[cat_data.colnames[0]].data.astype(int)
         # loop over the filters we want to keep to get
         # the number of the filter, n, and the name, b
         # (the final data has to have  2 * n_bands + 4 columns)
-        for n, b in enumerate(bands):
-            flux = cat_data[cat_data.colnames[1 + 2 * n]].astype(float)
-            flux_err = cat_data[cat_data.colnames[2 + 2 * n]].astype(float)
-            self.data[f"flux_{b.split('_')[1].split('.')[0]}"] = flux
-            self.flux_cols.append(f"flux_{b.split('_')[1].split('.')[0]}")
-            self.data[f"flux_err_{b.split('_')[1].split('.')[0]}"] = flux_err
-            self.flux_err_cols.append(f"flux_err_{b.split('_')[1].split('.')[0]}")
+        for n, b in enumerate(observed_filter_ids):
+            if b in all_filter_ids:
+                flux = cat_data[cat_data.colnames[1 + 2 * n]].data.astype(float)
+                flux_err = cat_data[cat_data.colnames[2 + 2 * n]].data.astype(float)
+                self.data[f"psf_{b}_flux"] = flux
+                self.flux_cols.append(f"psf_{b}_flux")
+                self.data[f"psf_{b}_flux_err"] = flux_err
+                self.flux_err_cols.append(f"psf_{b}_flux_err")
```

Comment:
Is it possible to make it configurable which flux column is used?  For the multiband catalogs, there are a few options:
- psf (what you're using now)
- kron
- segment
- various apertures (e.g., aper_02)

The one we picked out once upon a time to use was "segment", but it makes sense for something either in roman catalog handler or in roman catalog process to have a "flux_type" argument that is used to select the appropriate fields, instead of always using the PSF varieties as is currently done.

The PSF should be the best for stars but it's probably the worst choice for galaxies and therefore for photometric redshifts.